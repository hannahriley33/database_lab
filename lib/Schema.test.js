// Test that validate method returns the object with all fields cast
// Test that validate method throws an error if the object doesn't follow the schema

const Schema = require('./Schema');

const dog = {
  name: 'spot',
  age: 5,
  weight: '20 lbs'
};

describe('Schema', () => {
  const schema = new Schema({
    name: {
      type: String,
      required: true
    },
    age: {
      type: Number,
      required: true
    },
    weight: {
      type: String
    }
  });


  // // before each test
  // beforeEach(() => {
  //   nameValidator = new Validator('name', {
  //     type: String,
  //     required: true
  //   });
  // });



  it('correctly checks the entire object to see if it fits the schema', () => {
    expect(dog).toEqual({
      
      name: {
        type: String,
        required: true
      },
      age: {
        type: Number,
        required: true
      },
      weight: {
        type: String
      }
    });
  });
});
