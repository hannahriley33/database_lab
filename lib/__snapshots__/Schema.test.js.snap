// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schema throws on a bad schema 1`] = `"invalid schema >> Error: Missing required key >>name<<,Error: Cannot cast >>hi<< to Number"`;
